src/server/auth/server/api_server.go

```diff
@@ -989,7 +988,92 @@ func (a *apiServer) GetAuthToken(ctx context.Context, req *authclient.GetAuthTok
 		return nil, fmt.Errorf("error storing token for default pipeline user: %v", err)
 	}
 
-	return &authclient.GetAuthTokenResponse{token}, nil
+	return &authclient.GetAuthTokenResponse{
+		Token: token,
+		TTL:   req.TTL,
+	}, nil
+}
+
+func (a *apiServer) ExtendAuthToken(ctx context.Context, req *authclient.ExtendAuthTokenRequest) (resp *authclient.ExtendAuthTokenResponse, retErr error) {
+	a.LogReq(req)
+	defer func(start time.Time) { a.LogResp(req, resp, retErr, time.Since(start)) }(time.Now())
+	if !a.isActivated() {
+		return nil, authclient.NotActivatedError{}
+	}
+	if req.TTL == 0 {
+		return nil, fmt.Errorf("invalid request: ExtendAuthTokenRequest.TTL must be > 0")
+	}
+
+	// Only admins can extend auth tokens (for now)
+	tokenInfo, err := a.getAuthenticatedUser(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if !a.isAdmin(tokenInfo.Subject) {
+		return nil, fmt.Errorf("must be an admin to extend an auth token")
+	}
+
+	// Only let people extend tokens by up to two weeks (the equivalent of logging
+	// in again)
+	if req.TTL > defaultTokenTTLSecs {
+		return nil, fmt.Errorf("can only extend tokens by at most %d seconds", defaultTokenTTLSecs)
+	}
+
+	// The token must already exist. If a token has been revoked, it can't be
+	// extended
+	if _, err := col.NewSTM(ctx, a.etcdClient, func(stm col.STM) error {
+		var (
+			// The collections in which the request token might exist
+			tokens, pipelineTokens = a.tokens.ReadWrite(stm), a.pipelineTokens.ReadWrite(stm)
```

Comment:
This suggests pipeline tokens need to be renewed. I assume pps master handles this? The lines below suggest only `pipelineTokenInfo`s can be issued by admins. So we're issuing `pipelineTokenInfo`s to users of the vault plugin? In the world where non machine users need access to pachyderm via vault, this may need to change.